### PR TITLE
[Feature] remove EVM: add more logic for CitaExecutive:exec.[skip ci]

### PR DESCRIPTION
**Pull request by leeyr338**
- Add some logic to handle receipt.
- Use trie_db.
- Call Native contract.
- Handle fee and refund (hasn't finished yet) for charge model.

### CODE REVIEW

File: cita-executor/Cargo.toml [hunk 1]
```diff
@@ -48,5 +47,5 @@ rabbitmq = ["pubsub/rabbitmq"]
 zeromq = ["pubsub/zeromq"]
 kafka = ["pubsub/kafka"]
 privatetx = ["core-executor/privatetx"]
-evm-debug = ["core-executor/evm-debug"]
-evm-debug-tests = ["evm-debug", "core-executor/evm-debug-tests"]
+#evm-debug = ["core-executor/evm-debug"]
```
Comment (kaikai1024) [inline cita-executor/Cargo.toml]: Just delete.
All the files about evm will be deleted.

File: cita-executor/core/src/cita_executive.rs [hunk 2]
```diff
@@ -38,21 +42,21 @@ pub struct CitaExecutive<'a, B> {
...
 }
 
 impl<'a, B: DB + 'static> CitaExecutive<'a, B> {
     pub fn new(
         block_provider: Arc<BlockDataProvider>,
-        state_provider: State<B>,
+        state_db: State<B>,
         native_factory: &'a NativeFactory,
-        env_info: EnvInfo,
+        env_info: &'a EnvInfo,
```
Comment (kaikai1024) [inline cita-executor/core/src/cita_executive.rs]: Why change it to `&EnvInfo`?

File: cita-executor/core/src/contracts/native/factory.rs [hunk 3]
```diff
@@ -96,14 +96,12 @@ impl Default for Factory {
 
 #[derive(Debug)]
 pub enum NativeError {
-    OutOfGas,
```
Comment (kaikai1024) [inline cita-executor/core/src/contracts/native/factory.rs]: Why delete it?

File: cita-executor/core/src/libexecutor/auto_exec.rs [hunk 4]
```diff
@@ -42,9 +42,9 @@ lazy_static! {
 pub fn auto_exec(
```
Comment (kaikai1024) [inline cita-executor/core/src/libexecutor/auto_exec.rs]: Add a `FIXME` tag.

File: cita-executor/src/postman.rs [hunk 5]
```diff
@@ -379,7 +379,7 @@ impl Postman {
                     .filter(|hash| hash != &H256::default())
                     .collect();
 
-                let schedule = Schedule::new_v1();
+                let schedule = Schedule::default();
```
Comment (kaikai1024) [inline cita-executor/src/postman.rs]: Add a `v1` tag?

File: cita-executor/core/src/cita_executive.rs [hunk 6]
```diff
@@ -556,12 +708,12 @@ impl Default for VmExecParams {
...
 /// Information concerning the execution environment for vm.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct EnvInfo {
     /// The block number.
     pub number: BlockNumber,
-    /// The block author.
-    pub author: Address,
+    /// The fee refund address.
+    pub coin_base: Address,
```
Comment (kaikai1024) [inline cita-executor/core/src/cita_executive.rs]: Is the author and the coin_base same thing?

File: cita-executor/core/src/contracts/native/crosschain_verify.rs [hunk 7]
```diff
@@ -50,10 +50,10 @@ impl Contract for CrossChainVerify {
                 sig if sig == *GET_EXPECTED_BLOCK_NUMBER_FUNC => {
                     self.get_expected_block_number(params, data_provider)
                 }
-                _ => Err(NativeError::OutOfGas),
```
Comment (ouwenkg) [inline cita-executor/core/src/contracts/native/crosschain_verify.rs]: https://github.com/cryptape/cita/blob/develop/cita-executor/core/src/contracts/native/crosschain_verify.rs#L87

File: cita-executor/core/src/cita_executive.rs [hunk 8]
```diff
@@ -63,7 +67,7 @@ impl<'a, B: DB + 'static> CitaExecutive<'a, B> {
         &mut self,
         t: &SignedTransaction,
         conf: &BlockSysConfig,
-    ) -> Result<Receipt, ExecutionError> {
+    ) -> Result<ExecutedResult, ExecutionError> {
```
Comment (kaikai1024) [inline cita-executor/core/src/cita_executive.rs]: Maybe `ExecutionResult` & `ExecutionError` or`ExecutedResult` & `ExecutedError`?

File: cita-executor/core/src/cita_executive.rs [hunk 9]
```diff
@@ -85,6 +89,9 @@ impl<'a, B: DB + 'static> CitaExecutive<'a, B> {
         let base_gas_required = match t.action {
             Action::Create => tx_gas_schedule.tx_create_gas,
             _ => tx_gas_schedule.tx_gas,
+        } + match t.version {
+            0...2 => 0,
+            _ => t.data.len() * tx_gas_schedule.tx_data_non_zero_gas,
```
Comment (ouwenkg) [inline cita-executor/core/src/cita_executive.rs]: I do not clear about this, can you explains more?

**Comment by kaikai1024** [comment]
merge it first.

**Pull request merged**